### ISSUE TITLE
#Synthetic Monitoring #Glowroot

### ISSUE THREAD

**Issue by Aayushi-Agr** [glowroot/glowroot#739]
Hi ,
I have configured the synthetic monitoring for an application in glowroot and able to monitor it through Synthetic dashboard but it is continuously showing the same value (60,000 ms) in graph. I am not getting what is this value is all about.
Can anyone please help me out...
PFA image for reference

![Synthetic monitoring](https://user-images.githubusercontent.com/68536688/87917343-78444f00-ca92-11ea-9b76-c3cae561cb6a.png)

**Comment by santiago-flores-kenility** [comment]
Could you provide me your code? I can't make it work, I always receive Null Pointer exception, even when using the example:

import org.apache.http.impl.client.*;
import org.apache.http.client.methods.*;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.*;

import static org.openqa.selenium.support.ui.ExpectedConditions.*;

public class Example {

    public void test(CloseableHttpClient httpClient) throws Exception {
        // e.g.
        HttpGet request = new HttpGet("https://www.example.org");
        try (CloseableHttpResponse response = httpClient.execute(request)) {
            if (response.getStatusLine().getStatusCode() >= 400) {
                throw new Exception("Unexpected response status code: "
                        + response.getStatusLine().getStatusCode());
            }
        }
        // do not close the http client itself (it is re-used)
    }
}

**Comment by neil-stephen-mcgonigle** [comment]
Hi, did you resolve your issue? 

> Could you provide me your code? I can't make it work, I always receive Null Pointer exception, even when using the example:
> 
> import org.apache.http.impl.client._; import org.apache.http.client.methods._;
> 
> import org.openqa.selenium._; import org.openqa.selenium.support.ui._;
> 
> import static org.openqa.selenium.support.ui.ExpectedConditions.*;
> 
> public class Example {
> 
> ```
> public void test(CloseableHttpClient httpClient) throws Exception {
>     // e.g.
>     HttpGet request = new HttpGet("https://www.example.org");
>     try (CloseableHttpResponse response = httpClient.execute(request)) {
>         if (response.getStatusLine().getStatusCode() >= 400) {
>             throw new Exception("Unexpected response status code: "
>                     + response.getStatusLine().getStatusCode());
>         }
>     }
>     // do not close the http client itself (it is re-used)
> }
> ```
> 
> }
Did you resolve your issue, I am stuck here too.

**Comment by santiago-flores-kenility** [comment]
No, I'm using aws canary alarm right now, but would love to use glowroot instead!

**Comment by neil-stephen-mcgonigle** [comment]
> No, I'm using aws canary alarm right now, but would love to use glowroot instead!

thanks, for the quick reply appreciated.

**Comment by MikaMLehtonen** [comment]
Probably, for the first issue, the problem is Glowroot not having access to that site, firewall restriction?
The second issue, I got similar problem resolved by installing OpenJDK-devel package, so probably you (or Glowroot) are not using  JDK (that is needed?) but only JRE.

**Comment by MikaMLehtonen** [comment]
But now I started to wonder, in case of synthetic Java monitor, how to send clearance notification to Slack. I do get the alert when it's triggered, but after the problem is resolved, it would be nice to get notification about that also.

**Comment by Never-lab** [comment]
**60,000 ms** on a synthetic chart is almost always the **timeout** (60s) — the monitor didn’t get a successful faster response (firewall/DNS/auth, script error, missing JDK bits for Selenium, etc.), so every sample pins at the timeout. Fix connectivity/script first; the flat line is not a “baseline latency” metric.

Synthetic setup Qs → [Discussions → Q&A](https://github.com/glowroot/glowroot/discussions/categories/q-a) / wiki. Related Selenium notes also appear in older threads (#628 family).

@nowheresly — please close as support.
